src/main/java/org/cbio/gdcpipeline/reader/ClinicalStreamReader.java

```diff
@@ -0,0 +1,130 @@
+package org.cbio.gdcpipeline.reader;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.cbio.gdcpipeline.model.cbio.CBioClinicalDataModel;
+import org.cbio.gdcpipeline.model.cbio.PatientFileModel;
+import org.cbio.gdcpipeline.model.cbio.SampleFileModel;
+import org.cbio.gdcpipeline.model.gdc.nci.tcga.bcr.xml.clinical.brca._2.TcgaBcr;
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.item.ItemStreamException;
+import org.springframework.batch.item.file.ResourceAwareItemReaderItemStream;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Created by Dixit on 16/06/17.
+ */
+public class ClinicalStreamReader implements ResourceAwareItemReaderItemStream<CBioClinicalDataModel>{
+
+    private Resource resource;
+
+    private static Log LOG = LogFactory.getLog(ClinicalStreamReader.class);
+    private List<CBioClinicalDataModel> cBioClinicalDataModelList = new ArrayList<>();
+
+    @Value("#{jobExecutionContext[barcodeToSamplesMap]}")
+    private HashMap<String, List<String>> barcodeToSamplesMap;
+
+    @Value("${onco.code.tcga.brca}")
+    private String oncotree_code;
+
+    @Override
+    public CBioClinicalDataModel read() throws Exception {
+        if (!this.cBioClinicalDataModelList.isEmpty()){
+            return cBioClinicalDataModelList.remove(0);
+        }
+        return null;
+    }
+
+    private PatientFileModel setPatientFields(TcgaBcr tcgaBcr) {
```

Comment:
This is where you would use constructors for the concrete implementations of the abstract class